### PR TITLE
chore: remove `execa` dependency

### PR DESCRIPTION
**Pull request by aduh95**
The less dependencies we have, the better.

### CODE REVIEW

**Review by VoltrexKeyva — APPROVED** [review]

**Review by MoLow — APPROVED** [review]

**Comment by targos** [comment]
This seems to have broken `git node v8 major`: https://github.com/nodejs/node-v8/actions/runs/6080766368/job/16495239660

File: components/git/v8.js [hunk 1]
```diff
@@ -80,14 +80,16 @@ export function handler(argv) {
 
   options.execGitNode = function execGitNode(cmd, args, input) {
     args.unshift(cmd);
-    return execa('git', args, {
-      cwd: options.nodeDir,
-      ...input && { input }
+    return runAsync('git', args, {
+      spawnArgs: {
+        cwd: options.nodeDir,
+        ...input && { input }
```
Comment (targos) [inline components/git/v8.js:86]: `spawn` doesn't take an `input` option.

File: lib/update-v8/minorUpdate.js [hunk 2]
```diff
@@ -31,11 +31,14 @@ function getLatestV8Version() {
...
       const version = ctx.currentVersion;
       const currentV8Tag = `${version.major}.${version.minor}.${version.build}`;
-      const result = await execa('git', ['tag', '-l', `${currentV8Tag}.*`], {
-        cwd: ctx.v8Dir,
-        encoding: 'utf8'
+      const result = await runAsync('git', ['tag', '-l', `${currentV8Tag}.*`], {
+        captureStdout: true,
+        spawnArgs: {
+          cwd: ctx.v8Dir,
+          encoding: 'utf8'
```
Comment (targos) [inline lib/update-v8/minorUpdate.js:38]: nor does it take an encoding option

File: lib/update-v8/minorUpdate.js [hunk 3]
```diff
@@ -63,15 +66,17 @@ function doMinorUpdate() {
 }
 
 async function applyPatch(ctx, latestStr) {
-  const { stdout: diff } = await execa(
+  const diff = await runAsync(
     'git',
     ['format-patch', '--stdout', `${ctx.currentVersion}...${latestStr}`],
-    { cwd: ctx.v8Dir, encoding: 'utf8' }
+    { captureStdout: true, spawnArgs: { cwd: ctx.v8Dir, encoding: 'utf8' } }
```
Comment (targos) [inline lib/update-v8/minorUpdate.js:72]: ditto

File: lib/update-v8/minorUpdate.js [hunk 4]
```diff
@@ -63,15 +66,17 @@ function doMinorUpdate() {
...
   );
   try {
-    await execa('git', ['apply', '--directory', 'deps/v8'], {
-      cwd: ctx.nodeDir,
-      input: diff
+    await runAsync('git', ['apply', '--directory', 'deps/v8'], {
+      spawnArgs: {
+        cwd: ctx.nodeDir,
+        input: diff
```
Comment (targos) [inline lib/update-v8/minorUpdate.js:78]: ditto

**Pull request merged**